modules/library/metadata/src/main/java/org/geotools/factory/AbstractFactory.java

```diff
@@ -121,7 +119,7 @@
  * @version $Id$
  * @author Martin Desruisseaux
  */
-public class AbstractFactory implements Factory, RegisterableService {
+public class AbstractFactory implements Factory, RegistrableFactory {
```

Comment:
I wonder if we can repurpose our Factory interface and avoid multiple inheritance here?

Reply:
One thing should only extend the other if one thing _is_ the other. I can not decide whether that's the case here. Regarding implementation details, having one interface extend the other had these consequences:

* If `RegistrableFactory extends Factory` GeoTools users that want to be called back on registration would have to implement `getImplementationHints`.
* If `Factory extends RegistrableFactory` (leaving aside that then the more specific type would have a less specific name), lots of GeoTools classes would have to implement the registration functions even though they don't need them.
* If the interfaces were to be merged (e.g. into `Factory`), we had both downsides.

All situations could be improved by providing default implementations for some or all of these methods.

Your call.